rocketpy/simulation/flight_data_importer.py

```diff
@@ -0,0 +1,196 @@
+        self.filepath = filepath
+        self.columns_map = columns_map
+        self.units = units
+        self.interpolation = interpolation
+        self.extrapolation = extrapolation
+        self.separator = separator
+        self.encoding = encoding
+
+        self.columns = None
+        self.data = None
+        self.time_col = None
+
+        self.__handle_data()
+        self.__determine_time_column()
+        self.__create_attributes()
+
+    def __repr__(self):
+        return f"FlightDataImporter object: '{self.name}'"
+
+    def __handle_data(self):
+        raw_data = np.genfromtxt(
+            self.filepath,
+            delimiter=self.separator,
+            encoding=self.encoding,
+            dtype=np.float64,
+            names=True,
+            deletechars="",
+        )
+
+        # Store the column names
+        self.original_columns = raw_data.dtype.names
+        self.data = raw_data.view((np.float64, len(self.original_columns)))
+
+        # Create the columns map if necessary
+        if not self.columns_map:
+            self.columns_map = {col: col for col in self.original_columns}
+
+        # Map original columns to their new names (if mapped), otherwise keep as is
+        self.columns = [self.columns_map.get(col, col) for col in self.original_columns]
+
+    def __determine_time_column(self):
+        if self.columns_map:
+            for col, atr in self.columns_map.items():
+                if atr == "time":
+                    self.time_col = col
+                    return None
+        if "time" in self.original_columns:
+            self.time_col = "time"
+            return None
+        raise ValueError("Unable to determine the time column...")
+
+    def __create_attributes(self):
+        # Extract time values
+        try:
+            times = self.data[:, self.original_columns.index(self.time_col)]
+        except ValueError as e:
+            raise ValueError(
+                f"Unable to find column '{self.time_col}' in the header of the file."
+                + "The available columns are:"
+                + str(self.original_columns)
+            ) from e
+
+        created = []
+        for col, name in self.columns_map.items():
+            if name == "time":  # Handle time separately
+                setattr(self, name, times)
+                created.append(name)
+                continue
+
+            # Find the index of the current column in the filtered data
+            try:
+                col_idx = self.original_columns.index(col)
+            except ValueError:
+                warnings.warn(
+                    f"Unable to find column '{col}' in the header of the file. "
+                    + f"The attribute '{name}' won't be set. The available "
+                    + "columns are:"
+                    + str(self.original_columns),
+                    UserWarning,
+                )
+                continue
+
+            # Extract values for the current column
```

Comment:
The way you are iterating through the `columns_map` seem to inverted to the Juno3 example and to the `FLIGHT_LABEL_MAP`. The way its done here, the key is the name of the column in the file and the value is the name of the Flight attribute

Seems like it was intended to be the other way around (which I think is better also)

Reply:
You are right, I inverted that without noticing. 
The point is that the idea of "mapping" the columns is to get the corresponding Flight attribute for each column name of the dataset. I understand if may not be so beautiful, but I think if makes more sense to have something like `{"column_name": "Flight attribute name", ...}` than the opposite.

Reply:
The juno3 notebook was updated in another PR